### DISCUSSION TITLE
Manually trigger useAsyncList load method

### DISCUSSION THREAD

**Discussion by bdow** [adobe/react-spectrum#2698] [Q&A]
Hi all,

I was hoping to see if there was a smart way I could manually trigger the useAsyncList `load` method rather than having it automatically triggered. I want to wait until the user opens the ListBox before making the API call. Is this possible?

Thanks,
Brandon

**Comment by reidbarber** [comment]
At the moment, `load` will run on mount:

https://github.com/adobe/react-spectrum/blob/aa988c5e655abce21303853f62ed3f4f475e63a0/packages/%40react-stately/data/src/useAsyncList.ts#L315-L318

Maybe you could conditionally use a load function that returns an empty list until the ListBox is opened?

```js
 let list = useAsyncList({
    load: readyToLoad ? load : () => ({ items: [] })
  });
```

also you can use `list.reload()` to reload the data manually thereafter.
